File: delete_node_ll.go
package main

import "fmt"

type ListNode struct {
	Val  int
	Next *ListNode
}

func main() {
	var end *ListNode = &ListNode{4, nil}
	var third *ListNode = &ListNode{3, end}
	var head *ListNode = &ListNode{1, &ListNode{2, third}}
	deleteNode(third)
	for head != nil {
		fmt.Printf("%d->", head.Val)
		head = head.Next
	}
	fmt.Print("\n")
}

func deleteNode(node *ListNode) {
	var prev *ListNode;
	for node != nil && node.Next != nil{
		node.Val = node.Next.Val
		prev = node
		node = node.Next
	}
	prev.Next = nil
}

File: DeleteNodeLL.java
import structs.ListNode;

public class DeleteNodeLL {
    public static void main(String[] args){
        ListNode end = new ListNode(4);
        ListNode third = new ListNode(3, end);
        ListNode head = new ListNode(1, new ListNode(2, third));

        new DeleteNodeLL().deleteNode(third);
        while(third != null){
            System.out.print(third.val + "->");
            third = third.next;
        }
        System.out.print("\n");
    }

    public void deleteNode(ListNode node) {
        ListNode prev = node;
        while(node != null && node.next != null){
            node.val = node.next.val;
            prev = node;
            node = node.next;
        }
        prev.next = null;
    }
}
